modApiServer/test/org/aion/api/server/ApiAionTests.java

```diff
@@ -0,0 +1,398 @@
+    }
+
+
+    @Test
+    public void TestGetSync() {
+        System.out.println("run TestGetSync.");
+        AionImpl impl = AionImpl.inst();
+        ApiAionTests.ApiAionImpl api = new ApiAionTests.ApiAionImpl(impl);
+        SyncInfo sync = api.getSync();
+        assertNotNull(sync);
+        assertEquals(sync.done, impl.isSyncComplete());
+        if (impl.getInitialStartingBlockNumber().isPresent())
+            assertEquals(sync.chainStartingBlkNumber, (long) impl.getInitialStartingBlockNumber().get());
+        if (impl.getInitialStartingBlockNumber().isPresent())
+            assertEquals(sync.networkBestBlkNumber, (long) impl.getNetworkBestBlockNumber().get());
+        if (impl.getInitialStartingBlockNumber().isPresent())
+            assertEquals(sync.chainBestBlkNumber, (long) impl.getLocalBestBlockNumber().get());
+    }
+
+    @Test
+    public void TestGetTransactions() {
+        System.out.println("run TestGetTransactions.");
+        AionImpl impl = AionImpl.inst();
+        AionRepositoryImpl repo = AionRepositoryImpl.inst();
+        ApiAionTests.ApiAionImpl api = new ApiAionTests.ApiAionImpl(impl);
+
+        AionBlock parentBlk = impl.getBlockchain().getBestBlock();
+        byte[] msg = "test message".getBytes();
+        AionTransaction tx = new AionTransaction(repo.getNonce(Address.ZERO_ADDRESS()).toByteArray(),
+                Address.ZERO_ADDRESS(), Address.ZERO_ADDRESS(), BigInteger.ONE.toByteArray(),
+                    msg,100000, 100000);
+        tx.sign(new ECKeyEd25519());
+
+        AionBlock blk = impl.getAionHub().getBlockchain().createNewBlock(parentBlk,
+                Collections.singletonList(tx), false);
+
+        assertNotNull(blk);
+        assertNotEquals(blk.getTransactionsList().size(), 0);
+
+        impl.getAionHub().getBlockchain().add(blk);
+
+        assertTrue(blk.isEqual(api.getBlockByHash(blk.getHash())));
+        assertEquals(tx, api.getTransactionByBlockHashAndIndex(blk.getHash(), 0));
+        assertEquals(tx, api.getTransactionByBlockNumberAndIndex(blk.getNumber(), 0));
+        assertEquals(1, api.getBlockTransactionCountByNumber(blk.getNumber()));
+        assertEquals(1, api.getTransactionCountByHash(blk.getHash()));
+
+        blk = api.getBlockByHash(blk.getHash());
+
+        assertEquals(1, api.getTransactionCount(
+                blk.getTransactionsList().get(0).getFrom(), blk.getNumber()));
+        assertEquals(0, api.getTransactionCount(Address.EMPTY_ADDRESS(), blk.getNumber()));
+
+        assertEquals(tx, api.getTransactionByHash(tx.getHash()));
+
+    }
+
+    @Test
+    public void TestDoCall() {
+        System.out.println("run TestDoCall.");
+        AionImpl impl = AionImpl.inst();
+        AionRepositoryImpl repo = AionRepositoryImpl.inst();
+        ApiAionTests.ApiAionImpl api = new ApiAionTests.ApiAionImpl(impl);
+
+        byte[] msg = "test message".getBytes();
+
+        Address addr = new Address(Keystore.create("testPwd"));
```

Comment:
should clean the keystore file after test

Reply:
Indeed

Reply:
done